File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: list[int]) -> int:
        n=len(height)
        maxarea=0
        ht=0
        left=0
        right=n-1
        while left<right:
            ht=min(height[left],height[right])
            width=right-left
            area=width*ht
            maxarea=max(maxarea,area)
            if(height[left]<height[right]):
                left+=1
            else:
                right-=1
        return maxarea

File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int n=height.size();
        int left=0;
        int right=n-1;
        int maxarea=0;
        int ht=0;
        int width=0;
        int area=0;
        while(left<right){
            ht=min(height[left],height[right]);
            width=right-left;
            area=width*ht;
            maxarea=max(maxarea,area);
            if(height[left]<height[right]){
                left++;
            }
            else{
                right--;
            }
            }
            return maxarea;
        }
    
};